ts/src/renderer/phaser/classes/PhaserUnit.ts

```diff
@@ -1,250 +1,189 @@
-					PhaserAttributeBar.release(a);
-				});
-				attributes.length = 0;
-
-				// add attribute bars based on passed data
-				data.attrs.forEach((ad) => {
-					const a = PhaserAttributeBar.get(this);
-					a.render(ad);
-					attributes.push(a);
-				});
-			});*/
-
-		scene.events.on('update', this.update, this);
 	}
 
-	update (/*time: number, delta: number*/): void {
-
-		const unit = this.unit;
-		const container = unit._pixiContainer;
-		const texture = unit._pixiTexture;
-
-		if (unit._destroyed || container._destroyed) {
-
-			unit.off('follow', this.followListener);
-			this.followListener = null;
-
-			unit.off('stop-follow', this.stopFollowListener);
-			this.stopFollowListener = null;
-
-			unit.off('play-animation', this.playAnimationListener);
-			this.playAnimationListener = null;
-
-			unit.off('update-label', this.updateLabelListener);
-			this.updateLabelListener = null;
-
-			unit.off('hide-label', this.hideLabelListener);
-			this.hideLabelListener = null;
-
-			unit.off('render-attributes', this.renderAttributesListener);
-			this.renderAttributesListener = null;
-
-			unit.off('update-attribute', this.updateAttributeListener);
-			this.updateAttributeListener = null;
-
-			unit.off('render-chat-bubble', this.renderChatListener);
-			this.renderChatListener = null;
-			if (this.chat) this.chat.destroy();
-
-			// release all instantiated attribute bars
-			this.attributes.forEach((a) => {
+	private updateAttributeListener (data: {
+		attr: AttributeData;
+		shouldRender: boolean;
+	}): void {
+		console.log('PhaserUnit update-attribute', data); // TODO remove
+		const attributes = this.attributes;
+		let a: PhaserAttributeBar;
+		let i = 0;
+		for (; i < attributes.length; i++) {
+			if (attributes[i].name === data.attr.type) {
+				a = attributes[i];
+				break;
+			}
+		}
+		if (!data.shouldRender) {
+			if (a) {
 				PhaserAttributeBar.release(a);
-			});
-			this.attributes.length = 0;
-			this.attributes = null;
-
-			this.scene.events.off('update', this.update, this);
+				attributes.splice(i, 1);
+			}
+			return;
+		}
+		if (!a) {
+			a = PhaserAttributeBar.get(this);
+			attributes.push(a);
+		}
+		a.render(data.attr);
+	}
 
-			this.label = null;
-			this.sprite = null;
+	private renderChatListener (text): void {
+		console.log('create-chat', text); // TODO remove
+		if (this.chat) {
+			this.chat.showMessage(text);
+		} else {
+			this.chat = new PhaserChatBubble(this.scene, text, this);
+		}
+	}
 
-			this.destroy();
+	protected destroy (): void {
+		if (this.chat) this.chat.destroy();
 
-			return;
-		}
+		// release all instantiated attribute bars
+		this.attributes.forEach((a) => {
+			PhaserAttributeBar.release(a);
+		});
+		this.attributes.length = 0;
+		this.attributes = null;
 
-		this.x = container.x;
-		this.y = container.y;
+		this.label = null;
+		this.sprite = null;
 
-		if (this.chat) this.chat.update(this.x, this.y);
+		Object.keys(this.evtListeners).forEach((key) => {
```

Comment:
you should not copy code from subclass method bat rather call super.destroy() instead

Reply:
thanks! forgot to call super